kentik_api_library/examples/lables_crud_example.py

```diff
@@ -0,0 +1,77 @@
+# pylint: disable=redefined-outer-name
+"""
+Examples of using the typed labels API
+"""
+
+import os
+import sys
+import logging
+from typing import Tuple
+from kentik_api import kentik_api
+from kentik_api.public.device_label import DeviceLabel
+
+logging.basicConfig(level=logging.INFO)
+
+
+def get_auth_email_token() -> Tuple[str, str]:
+    try:
+        email = os.environ['KTAPI_AUTH_EMAIL']
+        token = os.environ['KTAPI_AUTH_TOKEN']
+        return email, token
+    except KeyError:
+        print('You have to specify KTAPI_AUTH_EMAIL and KTAPI_AUTH_TOKEN first')
+        sys.exit(1)
+
+def run_crud():
+    """
+    Expected response is like:
+
+    ### CREATE
+    CreateResponse(id=2794, name='apitest-label-1', color='#0000FF', user_id='144319', company_id='74333', devices=[], created_date='2020-12-02T14:39:46.686Z', updated_date='2020-12-02T14:39:46.686Z')
+
+    ### UPDATE
+    UpdateResponse(id=2794, name='apitest-label-one', color='#FF0000', user_id='144319', company_id='74333', devices=[], created_date='2020-12-02T14:39:46.686Z', updated_date='2020-12-02T14:39:46.686Z')
+
+    ### GET
+    GetResponse(id=2794, name='apitest-label-one', color='#FF0000', user_id='144319', company_id='74333', devices=[], created_date='2020-12-02T14:39:46.686Z', updated_date='2020-12-02T14:39:46.686Z')
+
+    ### DELETE
+    DeleteResponse(success=True)
+    """
+
+
+    email, token = get_auth_email_token()
+    client = kentik_api.for_com_domain(email, token)
+
+    print("### CREATE")
+    label = DeviceLabel.for_create("apitest-label-1", "#0000FF")
+    created = client.labels.create(label)
+    print(created.__dict__)
+    print()
+
+    print("### UPDATE")
+    label = DeviceLabel.for_update("apitest-label-one")
+    updated = client.labels.update(created.id, label)
+    print(updated.__dict__)
+    print()
```

Comment:
I'm not a fan of for_update/for_create methods. Especially, I don't think 'for_update' should ever be needed. The simpler approach would be something like this:
```suggestion
    label = DeviceLabel("apitest-label-1", "#0000FF")
    created = client.labels.create(label)
    
    created.name = "apitest-label-one"
    updated = client.labels.update(created)
```
If you're worried about passing unmodified fields in json payload, I think it is fine for now. If it becomes a problem, we can implement some mechanism to prevent that, but still, I think it can be hidden from the user.

Reply:
Disagree. 

Remember that DeviceLabel belongs to user-facing interface of our python library. It should provide guidance on how to correctly interact with api resources.
Sole __init__ constructor of DeviceLabel will surely work.
But now imagine you, as a user, want to create label. You start typing in your IDE:
label = DeviceLabel(
and a list of 8 params shows up.
How do you know, which params are required?  What happens, if you choose wrong? Exception? 400 bad request?

And later you want to update label.
You type again: 
label = DeviceLabel(
and again same list of 8 params shows up.
How do you know which ones are required this time? How do you know that this time some params can be provided but are optional?

Specialized constructors come to help the user use the library correctly.
This addresses the problem pointed out by Ted mentioning users struggle when a resource has 20 fields and among them 8 are required and they make mistakes.

Please comment on what are cons of this approach and what are alternatives that can provide the user similar level of support.

Reply:
In Python, you can make a parameter optional by providing a default value in function's signature. Other parameters are required as usual. Won't that be enough?

I don't think that creating DeviceLabel object to update an existing object would be a common use case. The workflow that I'd expect would be obtaining the object first, updating it, then passing it to the update method of the client.